dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/cloudera/manager/ClouderaClusterCPUChartTask.java

```diff
@@ -0,0 +1,153 @@
+  */
+  private static final String TS_CPU_QUERY_TEMPLATE =
+      "SELECT cpu_percent_across_hosts WHERE entityName = \"%s\" AND category = CLUSTER";
+
+  private final ObjectMapper objectMapper = new ObjectMapper();
+
+  private final int includedLastDays;
+  private final TimeSeriesAggregation tsAggregation;
+
+  public ClouderaClusterCPUChartTask() {
+    this(1, TimeSeriesAggregation.RAW);
+  }
+
+  public ClouderaClusterCPUChartTask(int includedLastDays, TimeSeriesAggregation tsAggregation) {
+    super(
+        String.format(
+            "cmf-cluster-cpu-%s-%s.jsonl",
+            includedLastDays, tsAggregation.toString().toLowerCase()));
+    this.includedLastDays = includedLastDays;
+    this.tsAggregation = tsAggregation;
+  }
+
+  @Override
+  protected Void doRun(
+      TaskRunContext context, @Nonnull ByteSink sink, @Nonnull ClouderaManagerHandle handle)
+      throws Exception {
+    CloseableHttpClient httpClient = handle.getHttpClient();
+    List<ClouderaClusterDTO> clusters = getClustersFromHandle(handle);
+
+    final String timeSeriesAPIUrl = buildTimeSeriesUrl(handle.getApiURI().toString());
+    try (Writer writer = sink.asCharSink(StandardCharsets.UTF_8).openBufferedStream()) {
+      for (ClouderaClusterDTO cluster : clusters) {
+        String cpuPerClusterQuery = buildQueryToFetchCPUTimeSeriesOnCluster(cluster.getId());
+        LOG.debug(
+            "Execute charts query: [{}] for the cluster: [{}].",
+            cpuPerClusterQuery,
+            cluster.getName());
+
+        URIBuilder uriBuilder = new URIBuilder(timeSeriesAPIUrl);
+        uriBuilder.addParameter("query", cpuPerClusterQuery);
+        uriBuilder.addParameter("desiredRollup", this.tsAggregation.toString());
+        uriBuilder.addParameter("mustUseDesiredRollup", "true");
+        uriBuilder.addParameter("from", buildISODateTime(this.includedLastDays));
+
+        try (CloseableHttpResponse chart = httpClient.execute(new HttpGet(uriBuilder.build()))) {
+          JsonNode jsonNode = objectMapper.readTree(chart.getEntity().getContent());
+          writer.write(jsonNode.toString());
+          writer.write('\n');
+        }
+      }
+    }
+
+    return null;
+  }
+
+  private List<ClouderaClusterDTO> getClustersFromHandle(@Nonnull ClouderaManagerHandle handle) {
+    List<ClouderaClusterDTO> clusters = handle.getClusters();
+    if (clusters == null) {
+      throw new MetadataDumperUsageException(
+          "Cloudera clusters must be initialized before CPU charts dumping.");
+    }
+    List<ClouderaClusterDTO> cpuClusters = new ArrayList<>();
+    for (ClouderaClusterDTO cluster : clusters) {
+      if (cluster.getId() == null) {
+        LOG.warn(
+            "Cloudera cluster id is null for cluster [{}]. " + "Skip CPU metrics for the cluster.",
+            cluster.getName());
+        // todo it's might be critical data for TCO calculation and we should fail the dump
```

Comment:
@dominik-klimczak please help to clarify this point.
It seems CPU usage is critical for TCO and if can't fetch it by Dumper, should we fail the entire dump, what do you think?

Reply:
I think we shouldn't fail. It's critical for TCO, but we can still present some useful data without it. We should work on validation in the service (Execution Summary in the report to highlight what is missing).